Assets/Scripts/Crafting/CraftingManager.cs

```diff
@@ -0,0 +1,150 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.VisualScripting;
+using UnityEngine;
+
+public class CraftingManager : MonoBehaviour
+{
+    public static CraftingManager Instance;
+
+    [SerializeField] Ingredient currentIngredient;
+    [SerializeField] SpriteRenderer[] drinkParts;
+    [SerializeField] Cup cup;
+
+    public Ingredient CurrentIngredient { get { return currentIngredient; }}
+
+
+    private void Awake()
+    {
+        if(Instance != null && Instance != this)
+        {
+            Destroy(this);
+        }
+        else
+        {
+            Instance = this;
+        }
+    }
+
+    [ContextMenu("Compare to recipes")]
+    bool CompareToRecipe()
+    {
+        print("comparing cup to recipes");
+        //determine base of the current drink
+        Ingredient baseIngredient = cup.baseIngredient;
+
+        Ingredient[] currentCupIngredients = cup.GetIngredients();
+
+        //find the recipe array specific to that base
+        RecipeDataSO[] recipesOfSameBase = RecipesHolder.Instance.ReturnBaseRecipeList(baseIngredient.Name);
+
+        //bool array that holds if the ingredient was already matched (for situations with 2+ of the same ingredients)
+        //including nulls!! ^^^^
+        bool[] matchingIngredients;
+
+        //for each recipe compare cups ingredients
+        foreach(RecipeDataSO recipe in recipesOfSameBase)
```

Comment:
Nesting so many loops is a big no-no. However, you need to deal with a problem with no simple answer. Talk to me and lets think about a better way to write this logic